core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -566,7 +566,10 @@ private[spark] class MemoryStore(
    * Release memory used by this task for unrolling blocks.
    * If the amount is not specified, remove the current task's allocation altogether.
    */
-  def releaseUnrollMemoryForThisTask(memoryMode: MemoryMode, memory: Long = Long.MaxValue): Unit = {
+  def releaseUnrollMemoryForThisTask(
+      memoryMode: MemoryMode,
+      memory: Long = Long.MaxValue,
+      releaseMemoryReally: Boolean = true): Unit = {
```

Comment:
This isn't a great var name... this also kind of feels hacky. Is there an easier way to refactor this without plumbing through a flag?

Reply:
Yeah, this modification is not very good.
Whether to add a new method just to release memory from `onHeapUnrollMemoryMap` or `offHeapUnrollMemoryMap`?

Reply:
I have updated this PR,  it doesn't look perfect, but it seems better than before.